src/test/java/core/basesyntax/HelloWorldTest.java

```diff
@@ -1,8 +1,127 @@
+    }
+
+    @Test
+    void register_validUser() {
+        User validUser = new User();
+        validUser.setAge(20);
+        validUser.setPassword("passwordValid");
+        validUser.setLogin("loginValid");
+
+        User result = registrationService.register(validUser);
+        assertNotNull(result);
+        assertEquals(validUser, storageDao.get("loginValid"));
+    }
+
+    @Test
+    void register_existingLogin_notOk() {
+        User firstUser = new User();
+        firstUser.setLogin("validLogin");
+        firstUser.setPassword("validPassword");
+        firstUser.setAge(20);
+        registrationService.register(firstUser);
+
+        User secondUserWithSameLogin = new User();
+        secondUserWithSameLogin.setLogin("validLogin");
+        secondUserWithSameLogin.setPassword("anotherPassword");
+        secondUserWithSameLogin.setAge(25);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(secondUserWithSameLogin);
+        });
+    }
+
+    @Test
+    void register_shortPassword_notOk() {
+        User userWithShortPassword = new User();
+        userWithShortPassword.setLogin("validLogin");
+        userWithShortPassword.setPassword("pwd");
+        userWithShortPassword.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithShortPassword);
+        });
+    }
+
+    @Test
+    void register_passwordIsNull_notOk() {
+        User userWithNullPassword = new User();
+        userWithNullPassword.setLogin("validLogin");
+        userWithNullPassword.setPassword(null);
+        userWithNullPassword.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithNullPassword);
+        });
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        User userWithShortLogin = new User();
+        userWithShortLogin.setLogin("log");
+        userWithShortLogin.setPassword("validPassword");
+        userWithShortLogin.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithShortLogin);
+        });
+    }
+
+    @Test
+    void register_loginIsNull_notOk() {
+        User userWithNullLogin = new User();
+        userWithNullLogin.setLogin(null);
+        userWithNullLogin.setPassword("validPassword");
+        userWithNullLogin.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithNullLogin);
+        });
+    }
+
+    @Test
+    void register_ageIsNull_notOk() {
+        User userWithNullAge = new User();
+        userWithNullAge.setLogin("validLogin");
+        userWithNullAge.setPassword("validPassword");
+        userWithNullAge.setAge(0);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithNullAge);
+        });
+    }
```

Comment:
You are testing the case where the age is 0, but the test name `register_ageIsNull_notOk` suggests that you are testing the case where the age is null. You should either update the test name to reflect the actual test case, or update the test case to match the test name.